compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -172,6 +172,13 @@ struct TokenCursor {
     // appended to the captured stream when
     // we evaluate a `LazyTokenStream`
     append_unglued_token: Option<TreeAndSpacing>,
+    // If `true`, skip the delimiters for `None`-delimited groups,
+    // and just yield the inner tokens. This is `true` during
+    // normal parsing, since the parser code is not currently prepared
```

Comment:
To work correctly it should be prepared though, cc https://github.com/rust-lang/rust/issues/67062.
(In other words, `skip_none_delims` should eventually become always `false`.)